src/email/mod.rs

```diff
@@ -696,10 +764,10 @@ mod test {
             .unwrap();
 
         assert_eq!(format!("{}", email),
-                   format!("To: <user@localhost>\r\nFrom: <user@localhost>\r\nCc: \"Alias\" \
-                            <cc@localhost>\r\nReply-To: <reply@localhost>\r\nSender: \
-                            <sender@localhost>\r\nDate: {}\r\nSubject: Hello\r\nX-test: \
-                            value\r\nMIME-Version: 1.0\r\nMessage-ID: \
+                   format!("Date: {}\r\nSubject: Hello\r\nX-test: value\r\nSender: \
+                            <sender@localhost>\r\nTo: <user@localhost>\r\nFrom: \
+                            <user@localhost>\r\nCc: \"Alias\" <cc@localhost>\r\nReply-To: \
+                            <reply@localhost>\r\nMIME-Version: 1.0\r\nMessage-ID: \
```

Comment:
Just needed to change the header order here. Maybe in the future, there should be an equality check function for tests, where the order of the generated headers is irrelevant.